test/test_b2http.py

```diff
@@ -137,6 +147,13 @@ def test_never_works(self):
                 pass
             self.assertEqual([call(1.0), call(1.5)], mock_time.mock_calls)
 
+    def test_too_many_requests_works_after_sleep(self):
```

Comment:
please make a test which shows how the two sleep mechanisms interact with each other.

If the first request had retry-after, and the second did not, but the third has a header again, what should happen?

If the first didn't have a header, second one has, and third one doesn't have, what should happen?